File: Prácticas/java/src/deepspace/SpaceCity.java
/**
 *
 * @author elena
 */

package deepspace;
import java.util.ArrayList;

public class SpaceCity extends SpaceStation {
    private SpaceStation base;
    private ArrayList<SpaceStation> collaborators;
    
    SpaceCity(SpaceStation base, ArrayList<SpaceStation> rest){
        super(base);
        this.base= base;
        this.collaborators= rest;
    }
    
    public ArrayList<SpaceStation> getCollaborators(){
        return collaborators;
    }
    
    @Override
    public float fire(){
        float f= super.fire();
        
        for(SpaceStation s: collaborators)
            f += s.fire();
        
        return f;
    }
    
    @Override
    public float protection(){
        float p= super.protection();
        
        for(SpaceStation s: collaborators)
            p += s.protection();
        
        return p;
    }
    
    @Override
    public Transformation setLoot(Loot l){
        super.setLoot(l);
        return Transformation.NOTRANSFORM;
    }
    
    @Override
    public SpaceCityToUI getUIversion(){
        return new SpaceCityToUI(this);
    }
}

File: Prácticas/ruby/lib/SpaceCity.rb
require_relative 'SpaceStation'
require_relative 'Transformation'
require_relative 'SpaceCityToUI'

module Deepspace
  class SpaceCity < SpaceStation
    attr_reader :collaborators
    
    def initialize(bas, rest)
      super(bas.name,bas.getSupplies, bas.nMedals, bas.weapons,bas.shieldBoosters,bas.hangar,bas.pendingDamage)
      @base= base 
      @collaborators= rest
    end
    
    def fire
      f= super
      
      for s in @collaborators
        f += s.fire
      end
      
      f
    end
    
    def protection
      p= super
      
      for s in @collaborators
        p += s.protection 
      end
      
      p
    end
    
    def setLoot(l)
      super
      Transformation::NOTRANSFORM
    end
    
    def getUIversion
      SpaceCityToUI.new(self)
    end
  end
end
